### DISCUSSION TITLE
Low-regularity constraints

### DISCUSSION THREAD

**Discussion by thartland** [coin-or/Ipopt#698] [General]
I just wanted to verify that Ipopt, which uses a Newton solver for the optimality conditions associated to a log-barrier subproblem, should not be used for problems where the Jacobian is _not_ continuously differentiable.

In a related note, are there any options to turn off the use of second order Hessian information and utilize a first order interior point solver in Ipopt?

**Comment by svigerske** [comment]
Yes, I think you could get into the convergence issues when functions are not twice continuously differentiable. That is, satisfying optimality conditions could be difficult.

There is no option in Ipopt to turn off the use of second order information. You can only tell it to approximate the Hessian instead of using an exact one (https://coin-or.github.io/Ipopt/SPECIALS.html#QUASI_NEWTON).
